app/routes/settings+/profile.photo.tsx

```diff
@@ -30,12 +30,14 @@ export const handle = {
 const MAX_SIZE = 1024 * 1024 * 3 // 3MB
 
 const PhotoFormSchema = z.object({
-	photoFile: z
-		.instanceof(File)
-		.refine(file => file.name !== '' && file.size !== 0, 'Image is required')
-		.refine(file => {
-			return file.size <= MAX_SIZE
-		}, 'Image size must be less than 3MB'),
+	photoFile: typeof window === 'undefined'
+		? z.undefined()
+		: z
+			.instanceof(File)
+			.refine(file => file.name !== '' && file.size !== 0, 'Image is required')
+			.refine(file => {
+				return file.size <= MAX_SIZE
+			}, 'Image size must be less than 3MB'),
```

Comment:
We definitely would still want this validation on the server. You may need to use Remix's `installGlobals`: https://github.com/remix-run/remix/blob/f2c4edfc28f7427189e67f32f12628e1c45b6b69/packages/remix-node/globals.ts#L45